sicp-doc/sicp-pict.scrbl

```diff
@@ -192,6 +197,85 @@ the first vect to the endpoint of the second vect.
   (vects->segments (list (make-vect 1 2) (make-vect 3 4) (make-vect 5 6) (make-vect 7 8)))]
 }
 
+@section{Colors, Pens, and Brushes}
+@defproc[(color-object? [v any/c]) boolean?]{
+  Defined as @racket[(is-a?/c color%)].
+}
+
+@defproc[(pen-object? [v any/c]) boolean?]{
+  Defined as @racket[(is-a?/c pen%)].
+}
+
+@defproc[(brush-object? [v any/c]) boolean?]{
+  Defined as @racket[(is-a?/c brush%)].
+}
+
+@deftogether[(@defproc[(new-color [color (or/c real? string? color-object?)]) color-object?]
+              @defproc[(new-color [r real?] [g real?] [b real?] [a (real-in 0 1) 1.0]) color-object?])]{
+  Create a color.
+}
+
+@defproc[(new-pen [color (or/c string? color-object?)]) pen-object?]{
+  Create a pen to draw lines and outlines.
+}
+
+@defproc[(new-brush [color (or/c string? color-object?)]) brush-object?]{
+  Create a brush to fill in areas.
+}
+
+@defproc[(new-stipple-brush [bm (or/c #f (is-a?/c bitmap%))]) brush-object?]{
+  Create a brush to fill in area with bitmap.
+}
+
+@defthing[black-color color-object?]{
+  Defined as @racket[(new-color "black")].
+}
+
+@defthing[white-color color-object?]{
+  Defined as @racket[(new-color "white")].
+}
+
+@defthing[black-pen pen-object?]{
+  Defined as @racket[(new-pen "black")].
+}
+
+@defthing[black-brush brush-object?]{
+  Defined as @racket[(new-brush "black")].
+}
+
+@defthing[transparent-brush brush-object?]{
+  Defined as @racket[(new-brush "transparent")].
+}
+
+@section{Syntactic Sugar}
+@defform[(echo painter-expr)]{
+  For debugging: print the @racket[paint-expr] then paint.
+  This makes it easy to see the expression that was used to produce an image.
+}
+
+@defform[(with-transformation transformation body ...)]{
+  Store the @racket[initial-matrix] of the drawing context given by @racket[current-dc].
```

Comment:
What is `initial-matrix`? It's not defined anywhere.

Reply:
It's a transformation matrix that converts logical coordinates to device coordinates. (https://docs.racket-lang.org/draw/dc___.html#%28meth._%28%28%28lib._racket%2Fdraw..rkt%29._dc~3c~25~3e%29._get-initial-matrix%29%29)

Reply:
Well, the question is, should _users_ of `sicp-pict` know about it? 

- It could be that this is just an implementation detail, and there's no need to mention it in the documentation to cause confusion.
- Or this detail is important, and in that case there should be more explanation, and it should link to the link that you gave to me. (the reason why I notice this issue in the first place is that `initial-matrix` is not typeset as a variable or linked, which usually indicates that it's incorrect)

Reply:
> It could be that this is just an implementation detail, and there's no need to mention it in the documentation to cause confusion.

I copied the comment of `with-tranformation` as its introduction, and I guess it's necessary to explain the usage of `with-tranformation` through `initial-matrix`.

> the reason why I notice this issue in the first place is that initial-matrix is not typeset as a variable or linked, which usually indicates that it's incorrect

You're right, I was just thinking that I should set font to `initial-matrix`, so I didn't realize it when I was browsing the web documentation.

> it should link to the link that you gave to me.

This looks good to me.

Reply:
I tried to add link it via `hyperlink`:

```racket
@hyperlink["https://docs.racket-lang.org/draw/dc___.html#%28meth._%28%28%28lib._racket%2Fdraw..rkt%29._dc~3c~25~3e% 29._get-initial-matrix%29%29"]{initial-matrix}
```

But I guess there must be a more appropriate way for scribble since it's a link to the Racket document?

Reply:
The idea is to use:

    `(require (for-label (only-in racket/draw get-initial-matrix)))`

in order to make the binding `get-initial-matrix` available at "label" (aka documentation rendering) time.

Then use `@racket[get-initial-matrix]` to generate the link.

When Scribble rendered the documentation for `racket/draw` it entered information on the identifers into a documentation database. Scribble will use that information to generate the link.